### DISCUSSION TITLE
Keda is not scaling beyond 1 job. Once a job is completed then only it's creating another job even if there are multiple events at the same time.

### DISCUSSION THREAD

**Discussion by chalokodekare** [kedacore/keda#3885] [Q&A / Need Help]
Hi Team,

I'm trying to create ScaledJob with Kafka scaler as the trigger. The requirement is to have one job for each Kafka msg. 

The same YAML was working a few days back but suddenly it stopped working. I've tried a lot of things but no luck so far. Any help would be really appreciated.

Btw, my Kafka has only 1 partition and I was still able to scale jobs more than 1 with the same Kafka configuration a few days back. 

**Requirement:** To have one job for each Kafka msg.
**Current Behaviour:**  One job is getting created at a time even if there are multiple msgs in the Kafka queue. Once the running job is completed then only a new job is getting created. 

Below is the configuration file. 
[sample.txt](https://github.com/kedacore/keda/files/10064527/sample.txt)

apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: keda-trigger-auth-kafka-credential
  namespace: default
spec:
  secretTargetRef:
  - parameter: sasl
    name: keda-kafka-secrets
    key: sasl
  - parameter: username
    name: keda-kafka-secrets
    key: username
  - parameter: password
    name: keda-kafka-secrets
    key: password
  - parameter: tls
    name: keda-kafka-secrets
    key: tls

---

apiVersion: keda.sh/v1alpha1
kind: ScaledJob
metadata:
  name: sparqengine-scaledjob-xs
spec:
  jobTargetRef:
    template:
      spec:
        containers:
        - name: sparq-engine
          image: 127.0.0.1:5000/sparq-engine:autoscaling
          imagePullPolicy: Always
          env:
            - name: SPRING_PROFILES_ACTIVE
              value: production
            - name: PORT
              value: "8080"
            - name: ENGINE_SIZE
              value: XS  
        restartPolicy: Never  
    parallelism: 1
    completions: 1
    activeDeadlineSeconds: 86400
    backoffLimit: 1
  pollingInterval: 5
  successfulJobsHistoryLimit: 0
  failedJobsHistoryLimit: 5
  minReplicaCount: 0
  maxReplicaCount: 5
  triggers:
  - type: kafka
    metadata:
      bootstrapServers: resource-name.servicebus.windows.net:9093
      consumerGroup: XS
      topic: private.abp.supply.sparqengine.startup
      lagThreshold: "1"
      offsetResetPolicy: earliest #should be same as consumer
    authenticationRef:
      name: keda-trigger-auth-kafka-credential

**Comment by chalokodekare** [comment]
I'm using 2.8.0 version of keda.

@zroubalik @ahmelsayed looking for some guidance.

**Reply by ritu-bhandari-by** [reply]
Does anybody know How to enable debug logs for KEDA ?
